### PR TITLE
Add tests for SpeechSynthesis pause() and resume()

### PR DESCRIPTION
**Pull request by foolip**
For https://bugs.chromium.org/p/chromium/issues/detail?id=679043

Spec bugs:
https://github.com/w3c/speech-api/issues/39
https://github.com/w3c/speech-api/issues/40

### CODE REVIEW

**Comment by foolip** [comment]
The tentative test passes in Firefox but fails in Chrome and Safari. I have verified that the state changes are async in Chrome's code though.

File: speech-api/SpeechSynthesis-pause-resume.tentative.html [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    assert_false(speechSynthesis.paused, 'paused state after speak()');
+
+    utter.onstart = t.step_func(() => {
+      utter.onstart = null;
+      assert_false(speechSynthesis.paused, 'paused state at start event');
+
+      speechSynthesis.pause();
+
+      // paused state changes async, right before the pause event
+      assert_false(speechSynthesis.paused, 'paused state after pause()');
```
Comment (csharrison) [inline speech-api/SpeechSynthesis-pause-resume.tentative.html]: Maybe add "synchronously" to the error message.

**Review by csharrison — APPROVED** [review]
LGTM

**Comment by guest271314** [comment]
@foolip See also "speak <previous utterance text> speak slash" is audio output of .speak() following two calls to .speak(), .pause() and .resume() https://bugs.chromium.org/p/chromium/issues/detail?id=797624

**Pull request merged**